### ISSUE TITLE
windows: not possible to re-index some tracks and albums

### ISSUE THREAD

**Issue by sakuramodokich** [harmonoid/harmonoid#322]
**Describe the bug:**

With the latest nightly version of Harmonoid, a few tracks and albums are not recognizable after re-index them.
After re-index the stats shows:
Tracks  411
Albums  116
Artists  207
Genres  9

Folder actually contains:
flac=347; wav=1; m3u=1; mp3=72; m4a=21; total: 442

**Desktop :**

OS: Windows 11
Version 0.2.9

**Comment by alexmercerind** [comment]
Hey,
Please attach logs.
Launch it like this from Command Prompt:

```
harmonoid.exe > logs.txt
```

You might need to launch it as administrator, based on your location & permission.

**Comment by sakuramodokich** [comment]
Hey @alexmercerind, I uploaded my log.txt on the link below:
https://drive.google.com/file/d/134Y_DJ2KdkrDtvMKONOd4lrvzn7Woted/view?usp=sharing

**Comment by alexmercerind** [comment]
Hey @sakuramodokich, thanks for the logs!
The log shows exactly 411 tracks (songs) being detected in your folder.

Currently we index following file extensions:

```dart
  'OGG',
  'OGA',
  'OGX',
  'AAC',
  'M4A',
  'MP3',
  'WMA',
  'WAV',
  'FLAC',
  'OPUS',
  'AIFF',
```

I see `M3U` is not in this list, I will add it. But please see if you have other formats in your folder, but not in this list.

Thanks again!

**Comment by sakuramodokich** [comment]
@alexmercerind I have 442 songs in my folder, but 32 songs are not shown in the menu even though they are readable.The following situations may also occur(they are in .flac format, which converted from .wav format):
<img width="751" alt="Snipaste_2022-09-03_16-26-24" src="https://user-images.githubusercontent.com/109240615/188262766-96278244-c76f-4e1c-a9d0-917894f6d9d3.png">

**Comment by alexmercerind** [comment]
@sakuramodokich, could you please mail me a sample of file which is not being shown (please don't change the file name) at alexmercerind@gmail.com.

<hr> 

You can use this command to get all the file extensions in your folder:

```cmd
Powershell -Command "Get-ChildItem . -Include *.* -Recurse | Select-Object Extension | Sort-Object -Property Extension -Unique"
```

**Comment by sakuramodokich** [comment]
@alexmercerind The file has been sent to you
**Output:**
`D:\AlbumDropHere> Powershell -Command "Get-ChildItem . -Include *.* -Recurse | Select-Object Extension | Sort-Object -Property Extension -Unique"

Extension
---------
. nakotanmaru)
. NEONA)
. NEONA) [KOTONOHOUSE Remix]
. nicamoq) - Single
. Tsugumi Nagahara) [WEB-FLAC]
. ローラーガール) - Single
.05 [SBCD-014] エイセイパレード [例大祭16]
.1
.2
.22] MEGAREX (Tatsunoshin x PIKASONIC x SARUKANI) — NANYAKORE {MRX-092} [WEB-FLAC]
.22] TVアニメ「無職転生 ～異世界行ったら本気だす～」Theme Song Collection [FLAC 96kHz／24bit]
.cue
.flac
.Holiday
.java
.jpg
.log
.lrc
.m3u
.m4a
.mp3
.pdf
.png
.tif
.wav`

**Comment by sakuramodokich** [comment]
I also wrote a java script to count by file type:
```
import java.io.File;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;
import java.util.Map.Entry;
import java.util.Set;
 
/**
 * 
 * 
 * @author Administrator
 *
 */
public class FileCountTest {
 
	// resultMap
	public static final Map<String, Integer> resultMap = new HashMap<String, Integer>();
    
    public static void main(String[] args) {
        getFileType("D:\\AlbumDropHere\\");
        
        // mapkeymap
        Set<Map.Entry<String, Integer>> entrySet = resultMap.entrySet();
        Iterator<Entry<String, Integer>> it = entrySet.iterator();
        while (it.hasNext()) {
            Entry<String, Integer> entry = it.next();
            System.out.println(entry.getKey() + "=" + entry.getValue());
        }
    }
    
    
    /**
     * 
     * @param path 
     */
    public static void getFileType(String path) {
        File file = new File(path);
        
        
        if (!file.exists()) {
            return ;
        }
        
        File[] fileList = file.listFiles();
        for (File fileTemp : fileList) {
        	
            if (fileTemp.isFile()) {
            	
                String endTemp = fileTemp.getName().substring(fileTemp.getName().lastIndexOf(".") + 1);
                Integer num = resultMap.get(endTemp);
                if (num == null) {
                    resultMap.put(endTemp, 1);
                } else {
                    resultMap.put(endTemp, num + 1);
                }
            }
            
            if (fileTemp.isDirectory()) {
            	
                String pathTemp = fileTemp.getAbsolutePath();
                getFileType(pathTemp);
            }
        }
    }
}
```
**Output:**
```
D:\AlbumDropHere> java type.java
jpg=60
log=2
flac=347
png=21
wav=1
m3u=1
tif=11
mp3=72
cue=2
pdf=1
java=1
m4a=21
lrc=2
db=1
```

**Comment by alexmercerind** [comment]
Hey @sakuramodokich, please [install the latest nightly](https://github.com/harmonoid/nightly/releases/tag/nightly).

Let me know if it fixes it (reply here whatever you find out).

![](https://user-images.githubusercontent.com/28951144/188263918-7d89b5fa-4016-4749-99b8-420990aa6c3c.png)

I look forward to release a hotfix with few other fixes already made. I will make sure to credit you in the changelog.

Thankyou.

**Comment by alexmercerind** [comment]
Please don't hesitate if the issue still persists.
Fixes like these are really important to get over the line soon.

**Comment by sakuramodokich** [comment]
Bug fixed(at least for me). Thank you ever so much!
<img width="750" alt="Snipaste_2022-09-03_22-22-32" src="https://user-images.githubusercontent.com/109240615/188274637-5b3a1b29-3702-44bf-aae6-db818a13c1b8.png">
<img width="754" alt="Snipaste_2022-09-03_22-17-50" src="https://user-images.githubusercontent.com/109240615/188274591-001b17c3-c38f-49b5-80ee-0c3fa976ca3a.png">

**Comment by sakuramodokich** [comment]
Also found an issue in the latest nightly: this button is not working
<img width="748" alt="Snipaste_2022-09-03_22-39-57" src="https://user-images.githubusercontent.com/109240615/188275605-c6d99705-dfca-442c-aa7e-3d61f806211f.png">

**Comment by alexmercerind** [comment]
> Also found an issue in the latest nightly: this button is not working.

Good catch!I will fix.

**Comment by alexmercerind** [comment]
Hi @sakuramodokich,

A stable version with the bug-fix has been released: [v0.3.0](https://github.com/harmonoid/harmonoid/releases/tag/v0.3.0).
Many other important things are also fixed. 

Thanks for the descriptive reports & file samples. Hope you have fun!
